develop/tutorials/articles/02-from-liferay-6-to-liferay-7/07-upgrading-plugins-to-liferay-7/07-upgrading-application-upgrade-and-verifier-processes.markdown

```diff
@@ -0,0 +1,426 @@
+
+    After you make changes to your module you might increment the module
+    version, however the database schema version may remain untouched, and
+    therefore have a different value. If your module's database schema is a 
+    different version than your module's bundle version, you can specify it in 
+    the BND file.
+    
+    Add the`Liferay-Require-SchemaVersion` property to your `bnd.bnd` file to 
+    describe that the schema version of the module uses a different version than 
+    the module's bundle version.
+    
+    For example, the [Chat App](https://github.com/liferay/liferay-portal/blob/0d69917/modules/apps/chat/chat-service/bnd.bnd)
+    specifies the bundle version and schema version below:
+
+        Bundle-Version: 1.0.2
+        
+        ...
+        
+        Liferay-Require-SchemaVersion: 1.0.0
+
+    If no `Liferay-Require-SchemaVersion` value is found, the `Bundle-Version`
+    header will be used.
+    
+    Note that once you make an upgrade to your app, it is best practice to 
+    specify your schema version with the `Liferay-Require-SchemaVersion` 
+    property from that point on. This way, if the bundle version and schema 
+    version ever differ, your database schema version will always be correct.
+
+3.  Convert your `upgrade` class into an OSGi component, using the `@Component`
+    token and implement the `UpgradeStepRegistrator` interface, as shown in the
+    example below:
+
+        @Component(
+                immediate = true,
+                service = UpgradeStepRegistrator.class
+        )
+        public class CalendarServiceUpgrade implements UpgradeStepRegistrator
+
+4.  Remove the intermediate classes that wrapped the internal steps, i.e the
+    `public class UpgradeProcess_1_0_1 extends UpgradeProcess...`.
+
+    Following the prior upgrade process in 6.2, you had to define the property
+    `upgrade.processes`, a list of `UpgradeProcesses` representing the different
+    upgrades for a specific version of your module, in your
+    `portal-ext.properties`.
+
+    For instance, the code below shows the previous process for upgrading
+    Calendar-service module from v1.0.0 to 1.0.1 and then to 1.0.2.
+
+        upgrade.processes=
+            com.liferay.calendar.hook.upgrade.UpgradeProcess_1_0_0,
+            com.liferay.calendar.hook.upgrade.UpgradeProcess_1_0_1,
+            com.liferay.calendar.hook.upgrade.UpgradeProcess_1_0_2
+
+    Each step between versions was represented by a single class extending
+    `UpgradeProcess`, using a method called `doUpgrade`. This method was
+    responsible for executing the internal steps to update the database to that
+    concrete version. A method `getThreshold` was provided also to specify the
+    schema version where the upgrade starts.
+
+    Whenever you needed another internal step, you added another
+    `upgrade(new UpgradePortletPreferences());` etc. after the existing ones.
```

Comment:
@mdelapenya, @mwilliams2014 It's not clear what is meant by an "internal step". Before this paragraph, it would be helpful to show an example of an `UpgradeProcess` extension class that has multiple internal steps. I'm not sure at this point whether the example line `upgrade(new UpgradePortletPreferences());` is something a developer would insert into most `UpgradeProcess` extensions or whether it's for a particular extension.